compiler/noirc_frontend/src/parser/parser/function.rs

```diff
@@ -76,16 +79,32 @@ fn function_modifiers() -> impl NoirParser<(bool, ItemVisibility, bool)> {
         })
 }
 
+pub(super) fn numeric_generic() -> impl NoirParser<UnresolvedGeneric> {
+    keyword(Keyword::Let)
+        .ignore_then(ident())
+        .then_ignore(just(Token::Colon))
+        .then(parse_type())
+        .map(|(ident, typ)| UnresolvedGeneric::Numeric { ident, typ })
+}
+
+pub(super) fn generic_variable() -> impl NoirParser<UnresolvedGeneric> {
+    ident().map(UnresolvedGeneric::Variable)
+}
+
+pub(super) fn generic() -> impl NoirParser<UnresolvedGeneric> {
+    numeric_generic().or(generic_variable())
+}
+
 /// non_empty_ident_list: ident ',' non_empty_ident_list
 ///                     | ident
 ///
 /// generics: '<' non_empty_ident_list '>'
 ///         | %empty
-pub(super) fn generics() -> impl NoirParser<Vec<Ident>> {
-    ident()
+pub(super) fn generics() -> impl NoirParser<UnresolvedGenerics> {
+    generic()
         .separated_by(just(Token::Comma))
         .allow_trailing()
-        .at_least(1)
+        // .at_least(1)
```

Comment:
I was going to comment on this being commented code and us not wanting to allow an empty `<>` list but apparently rust allows this so maybe we should remove it.

Reply:
I removed it